### DISCUSSION TITLE
Blacklist Additions

### DISCUSSION THREAD

**Discussion by Ichisich** [RamonUnch/AltDrag#364] [General]
https://github.com/RamonUnch/AltSnap/issues/352#issuecomment-1403340737

Some Windows 10 (maybe 11) specific ui element classes:

`|TaskListThumbnailWnd`
- taskbar thumbnails

`|NotifyIconOverflowWindow` 
- tray icon overflow panel

`*|Windows.UI.Core.CoreWindow` 
- calendar
- notification panel (win+a)
- task view (win+tab)
- whiteboard (win+w)
- screen clipping (win+shift+s)
- probably a lot more

`|NativeHWNDHost`
- media panel

`*|Xaml_WindowedPopupClass`
- some obscure modern styled kontext menus

`Input Flyout|Shell_InputSwitchTopLevelWindow`
- keyboard language layout switcher (win+space)

Elements I have no idea how to blacklist without kneecapping other important programs:
- On-screen keyboard:
ahk_class ApplicationFrameWindow
ahk_exe explorer.exe
ahk_pid 6492
ahk_id 66678

- Win+v paste panel:
ahk_class ApplicationFrameWindow
ahk_exe explorer.exe
ahk_pid 6492
ahk_id 2820140

- Xbox Game Bar (win+g)
ahk_class ApplicationFrameWindow
ahk_exe explorer.exe
ahk_pid 6492
ahk_id 3475680

**Comment by RamonUnch** [comment]
If there are other `explorer.exe:*|ApplicationFrameWindow` combinations that should not be blacklisted then unfortunately we cannot add them, the rest can be safely added though.

**Comment by Ichisich** [comment]
I will use `explorer.exe:*|ApplicationFrameWindow` the next days and see if something gets fishy. 
The same with the other classes.

**Comment by Ichisich** [comment]
So far, I only found one additional application with the combo:
`explorer.exe:*|ApplicationFrameWindow`
Voice typing (win+h).

The GUI consists of only one bar on top of the screen and doesn't look like it's supposed to be movable, but it is.
I would wager Microsoft forgot to disable dragging.
![voiceTyping](https://user-images.githubusercontent.com/31080003/215808418-55261d3b-dc0b-4fdd-93f5-ab3ef0ec92e1.png)

`*|Windows.UI.Core.CoreWindow`
Seems to do most of the heavy lifting. Almost all natively unmovable/ unresizable ui elements in Windows 10 have this class.
It even deprecated older entries AltSnap uses like `StartMenuExperienceHost.exe` and `SearchApp.exe`.

**Comment by Ichisich** [comment]
Some days later and I can claim all listed entries should be safe to use.
I even would include `explorer.exe:*|ApplicationFrameWindow`.

**Comment by RamonUnch** [comment]
Very nice.
I think you could start a Discussion in the Discussion Thread just about blacklist additions where people could contribute to a common knowledge about what *should* be blacklisted for everyone and also blacklist entries for more specific needs, whenever it is not trivial.

Your blacklist entries will be added in the next release.
I also think I should make a wiki section about all blacklist items wad what they do, like this people can eventually remove the entries they do not like (if this were to happen) also it would help people to see if something is missing.

EDIT: I converted the issue into a discussion

**Comment by mbartlett21** [comment]
@RamonUnch 
There is also the secondary taskbar(s) in Windows 10: `|Shell_SecondaryTrayWnd`.

(I'd assume it would be the same for XP/Vista/7/8)

**Comment by RamonUnch** [comment]
`mstsc.exe` and `msrdc.exe` also will be in the future default processes blacklist. #474
